libdd-profiling/src/internal/profile/mod.rs

```diff
@@ -297,6 +393,29 @@ impl Profile {
             Self::backup_period(period),
             Self::backup_sample_types(sample_types),
             None,
+            None,
+        )
+    }
+
+    /// Tries to create a profile with the given period and sample types. The
+    /// [`StringId2`]s should belong to the provided [`ProfilesDictionary`].
+    #[inline(never)]
+    #[cold]
```

Comment:
why?

Reply:
How often do you make new Profile objects? I'd consider once per minute pretty cold, wouldn't you?